File: compositor/src/main/java/org/westford/launch/LifeCycleSignals.java
package org.westford.launch;


import org.westford.Signal;
import org.westford.Slot;
import org.westford.compositor.core.events.Activate;
import org.westford.compositor.core.events.Deactivate;
import org.westford.compositor.core.events.Start;
import org.westford.compositor.core.events.Stop;

public interface LifeCycleSignals {
    Signal<Activate, Slot<Activate>> getActivateSignal();

    Signal<Deactivate, Slot<Deactivate>> getDeactivateSignal();

    Signal<Start, Slot<Start>> getStartSignal();

    Signal<Stop, Slot<Stop>> getStopSignal();
}

File: compositor/src/main/kotlin/org/westford/launch/LifeCycleSignals.kt
package org.westford.launch

import org.westford.Signal
import org.westford.compositor.core.events.Activate
import org.westford.compositor.core.events.Deactivate
import org.westford.compositor.core.events.Start
import org.westford.compositor.core.events.Stop

interface LifeCycleSignals {
    val activateSignal: Signal<Activate>

    val deactivateSignal: Signal<Deactivate>

    val startSignal: Signal<Start>

    val stopSignal: Signal<Stop>
}
